ivy/functional/frontends/torch/tensor.py

```diff
@@ -1371,6 +1371,11 @@ def mm(self, mat2):
     def square(self):
         return torch_frontend.square(self._ivy_array)
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+    def square_(self):
+        self.ivy_array = torch_frontend.square(self._ivy_array).ivy_array
```

Comment:
Use `ivy.astype()` here, rather than using `.ivy_array`

Reply:
Hello @zaeemansari70, I would greatly appreciate your guidance. Could you kindly clarify the usage of `ivy.astype()`? My intention was to follow the pattern of other in-place methods by utilizing `self.ivy_array` for in-place operations. Thank you kindly. 🙂